base/base/Typelists.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
```

Comment:
File should be named `TypeLists`?

Reply:
The only reason I named this header "Typelists" is that you do not need to switch to capital letters twice.
Google search for "typelist" (as a single word) produces lots of results so I guess a "typelist" is also fine.
I can change to "TypeLists" if you think it would be better, though.

Reply:
Ok, we can leave it at is.